crates/client/mapping-sync/src/sync_blocks.rs

```diff
@@ -40,6 +40,7 @@ where
 
                         // Success, we write the Starknet to Substate hashes mapping to db
                         let mapping_commitment = mc_db::MappingCommitment {
+                            block_number: digest_starknet_block.header().block_number,
                             block_hash: substrate_block_hash,
                             starknet_block_hash: digest_starknet_block_hash.into(),
                             starknet_transaction_hashes: digest_starknet_block
```

Comment:
why do we recover all the digest?

Reply:
This isn't my code sry, couldn't tell you. You should see with @antiyro